Makefile

```diff
@@ -28,10 +28,23 @@ else
   ponydir ?= $(prefix)/lib/pony/$(tag)
 endif
 
+# Use clang by default
+ifndef CC
+  ifneq (,$(shell clang --version 2>&1 | grep 'clang version'))
+    CC = clang
+    CXX = clang++
+  endif
+else ifeq ($(CC), cc)
```

Comment:
So if I have cc set to gcc. And I say CC=cc, we are going to override as clang? 

My first thought is "that sounds like a bad idea". However, I am guessing that the default behavior on many distros is to do CC=cc right?

Which is probably gcc.

So basically, we need to tell people "you need to explicitly set CC to gcc and CXX to g++ if you want to use gcc." I think that should be noted in BUILD.md.

I would find it surprising as a user.

Reply:
I think it also makes sense to note in a comment here that overriding CC being set to gcc is intentional and not an oversite as some folks might think it is a bug rather than intentional give the comment is "clang as default".

Reply:
OK, added a comment to that effect and notes in INSTALL.md and BUILD.md.